.github/workflows/release-reproducible-smart-contract.yaml

```diff
@@ -21,31 +21,32 @@ jobs:
         uses: actions/checkout@v4
         with:
           submodules: recursive
-      # Uses the `docker/login-action` action to log in to the Container registry.
-      - name: Log in to the Container registry
-        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
         with:
-          registry: ${{ env.REGISTRY }}
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
+          toolchain: ${{ env.RUST_VERSION }}
+          target: wasm32-unknown-unknown
+      - run: docker build -f ./reproducible/build.Dockerfile --build-arg source_image="rust:$RUST_VERSION" -t "concordium/$SERVICE_NAME:$RUST_VERSION" ./reproducible
+      - name: Test smart contract
+        working-directory: ./reproducible/example-contract
+        run: |
+            CARGO_CONCORDIUM="cargo run --manifest-path $GITHUB_WORKSPACE/cargo-concordium/Cargo.toml --target x86_64-unknown-linux-gnu -- concordium"
+            $CARGO_CONCORDIUM build --verifiable "concordium/$SERVICE_NAME:$RUST_VERSION"
+            $CARGO_CONCORDIUM verify-build --module ./concordium-out/module.wasm.v1 --source ./concordium-out/module.wasm.v1.tar
       - name: Extract version from git tag
```

Comment:
Maybe move this step up and check the version from the tag before building the image, since it would fail a lot faster then.